com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -1077,16 +1095,9 @@ internal void Initialize(NetworkManager networkManager)
         /// </summary>
         internal void Shutdown()
         {
-            LocalClient.IsApproved = false;
-            LocalClient.IsConnected = false;
-            ConnectedClients.Clear();
-            ConnectedClientIds.Clear();
-            ConnectedClientsList.Clear();
+
```

Comment:
Not a critique, just curious why these (and ProcessSendQueues) needed to be moved, just so I can understand the changes you made better.

Reply:
Ahh... yeah this is because the `DisconnectRemoteClient(clientId)` invocation below on line 1132 can invoke OnClientDisconnectFromServer which a user could send out messages during that time that would not get sent.
`DisconnectRemoteClient`:
```
        internal void DisconnectRemoteClient(ulong clientId)
        {
            MessageManager.ProcessSendQueues();
            OnClientDisconnectFromServer(clientId);
        }
```
It processes the send queue prior to invoking `OnClientDisconnectFromServer(clientId);`. However, on line 995 the `ClientDisconnectedMessage` is being sent to any remaining connected clients. This means the last client to be disconnected would not get the `ClientDisconnectedMessage` if a host or server was shutting down.